### PR TITLE
Changed InputMethod to an abstract class and added an option to use a…

### PR DESCRIPTION
**Pull request by jwong101**
…nalog or digital input for the arm and intake arm

### CODE REVIEW

**Comment by J0hnCena** [comment]
Don't merge yet, I'm going to redesign the input method abstract class and add partnerxbox input during the meeting. I'll also add test cases for both input types.

**Comment by LoadingPleaseWait** [comment]
That's a pretty big diff, man.

**Comment by jwong101** [comment]
Are you sure this diff can fit inside the repo?

**Comment by LoadingPleaseWait** [comment]
It's a pretty gargantuan diff but this repo is capable of enlarging.

File: test/org/impact2585/frc2016/tests/InputMethodTest.java [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+		analogMoveIntakeTowardsBot = 1;
+		Assert.assertTrue(input.moveIntake() == 1);
+		
+		//tests if the input arms don't move if Harris presses both triggers
+		analogMoveIntakeAwayFromBot = 1;
+		Assert.assertTrue(input.moveIntake() == 0);
+		
+		//tests if the input arms don't move if one of the analog inputs are negative
+		analogMoveIntakeAwayFromBot = -1;
+		Assert.assertTrue(input.moveIntake() == 0);
```
Comment (LoadingPleaseWait) [inline test/org/impact2585/frc2016/tests/InputMethodTest.java:107]: That's interesting that you're testing for that. If one of the triggers gave a negative value, that would mean that something has gone horribly wrong.

**Pull request merged**